### DISCUSSION TITLE
[2023-04-03] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by wdanilo** [enso-org/enso#6183] [Weekly Demo Day]
On my end, I was mainly working with different teams, but also I implemented the first version of GUI-part of the list-editor, a widget allowing you to insert, re-order, and remove elements interactively. For now, re-ordering and removing is implemented only. The rest of the operations will be implemented this week. In order to implement it, a lot of bugs/issues were fixed, including new event management system.

https://user-images.githubusercontent.com/1623053/229486289-be45c689-d333-4371-9dd4-81f2057fa579.mov

**Comment by farmaazon** [comment]
Merged the Vector Editor Stub. Progress of the next task was tine, due to excessive time spent on other activity (triaging issues, reviewing and testing PRs, planning etc.).

**Comment by MichaelMauderer** [comment]
I was working on 
* finishing the execution mode dropdown and its integration in current graph editor top bar. 
* starting to investigate the issue around the disappearing visualizations. No solution so far.
Overall progress that week was due to several build issues (#6145, #6091) I encountered which required me to spend a lot of time on rebuilding our project from scratch.


https://user-images.githubusercontent.com/1428930/229518158-2c0c479c-d5fb-4949-967c-725a244ea6cb.mp4

**Comment by Frizi** [comment]
I was working on widget tree on nodes. Progress on actual widget tree implementation was small due to a large number of external issues.

- Finished remaining work on alignment in manual layouts and shapes #6087
- Removed `content_origin` layout property, allowing for overflowing layout elements (used by dropdown widget) #6155 
- Fixed a bug with missing view hierarchy updates for size-only changes (part of content_origin PR).
- Tracked down and reported a spirv-cross bug that affected all shape shaders #6164
- Used fixed automatic layout to position label and dropdown widgets correctly on the node available area.
![image](https://user-images.githubusercontent.com/919491/229521370-1ad34091-a47d-42ca-b8fb-ef949758abb5.png)

Next up, I'll be adding back proper port behavior (edge dragging) to the widgets. The widget tree building also needs more work in order to avoid unnecessary updates.

**Comment by vitvakatu** [comment]
Last week I finished support for adding/removing _execution context switch expressions_ on nodes (https://github.com/enso-org/enso/pull/6121) and also implemented hiding these expressions (https://github.com/enso-org/enso/pull/6136, the PR is on review).

Also, I fixed a regression introduced by a new cursor-aware behavior of the CB: https://github.com/enso-org/enso/pull/6120

https://user-images.githubusercontent.com/6566674/228579581-f6a4c214-af56-40f8-af8e-af102cf476b3.mp4

**Comment by Procrat** [comment]
Last week, I made our tooltips behave a bit better: the text is now vertically centered and they appear after a delay of half a second.
![](https://user-images.githubusercontent.com/607786/228164744-5a3ae878-11ab-4cdd-8fdf-93bdf4fe84fe.gif)

I also defined a second version of an icon for a button we're going to place next to nodes soon to enable/disable the output context.
![](https://user-images.githubusercontent.com/607786/229530416-cda31aa9-9906-440e-bb24-284048c0a37d.png)

**Comment by mwu-tow** [comment]
Applying CR feedback and significant improvements to #6077.
Discussing #5239 with Paweł and started work on that. 
Ongoing support for CI and giving advice when something fails.
Fixing a regression in enso-formatter caused by introduction of symlinks in the dashboard-generated files: #6172.

**Comment by kazcw** [comment]
- Finished #6048 (memory management within GPU buffers, and *Layer Symbol Partition* interface).

[vokoscreenNG-2023-03-29_21-13-43.webm](https://user-images.githubusercontent.com/1047859/229537671-09555bd3-8f7e-47e3-b1d2-62baf0db34d4.webm)

*Layer Symbol Partitions* allow adding and removing instances of a symbol at specific depth-order classes (partitions). All partitions are allocated in the same set of buffers, and they are drawn together in one draw call.

- Started debugging hardware-specific performance problems: #6030.
